dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/AuthorizationRestRepository.java

```diff
@@ -154,12 +155,36 @@ public AuthorizationRest findOne(Context context, String id) {
 
         Context context = obtainContext();
 
-        BaseObjectRest obj = utils.getBaseObjectRestFromUri(context, uri);
-        if (obj == null) {
-            return null;
+        EPerson currUser = context.getCurrentUser();
+
+        // get the user specified in the requested parameters, can be null for anonymous
+        EPerson user = getUserFromRequestParameter(context, epersonUuid);
+        if (ObjectUtils.notEqual(currUser, user)) {
+            // Temporarily change the Context's current user in order to retrieve
+            // authorizations based on that user
+            context.switchContextUser(user);
+        }
+
+        List<Authorization> authorizations = findAuthorizationsForUri(context, user, uri, featureName);
+
+        if (currUser != user) {
+            // restore the real current user
+            context.restoreContextUser();
         }
+        return converter.toRestPage(authorizations, pageable, utils.obtainProjection());
+    }
+
+    @PreAuthorize("#epersonUuid==null || hasPermission(#epersonUuid, 'EPERSON', 'READ')")
+    @SearchRestMethod(name = "objects")
+    public Page<AuthorizationRest> findByObjects(@Parameter(value = "uuid") List<String> uuidList,
+            @Parameter(value = "type") String type,
```

Comment:
Same deal here, I'd also make type required as a request without a type isn't going to be meaningful anyway. However, this will be handled cleanly as you included a check for this later on in the code. This means that implicitely this parameter is required, so we're better off making it explicitely required by setting it here.